lib/langchain/vectorsearch/elasticsearch.rb

```diff
@@ -0,0 +1,115 @@
+# frozen_string_literal: true
+module Langchain::Vectorsearch
+  class Elasticsearch < Base
+    attr_accessor :es_client, :index_name, :options
+
+    def initialize(url:, index_name:, llm:, api_key: nil, es_options: {})
```

Comment:
Please add the necessary ENV placeholders to `.env.example`, like `ELASTICSEARCH_URL=`.